### PR TITLE
Use presentation compiler when scaladocing.

### PR DESCRIPTION
**Pull request by vigdorchik**
Don't run unneeded phases when generating scaladoc.
Review by @VladUreche and @dragos.

### CODE REVIEW

**Comment by VladUreche** [comment]
LGTM 
Since there are two reviewers assigned, let's wait for @dragos to comment too.

**Comment by dragos** [comment]
:+1: LGTM!

**Comment by hubertp** [comment]
Is there any chance to merge forScaladoc and forInteractive?

**Comment by hubertp** [comment]
Ok. In that case you should also remove !forScaladoc when adapting ConstantType in Typers.scala.

**Comment by VladUreche** [comment]
I don't think `forScaladoc` implies `forInteractive` -- running `./scaladoc X.scala`we have `forScaladoc=true` and `forInteractive=false`.

**Comment by hubertp** [comment]
Ok, but with this change when running interactive compiler you have forScaladoc and forInteractive both true (so actually my previous comment was wrong, you should remove !forInteractive).

File: src/compiler/scala/tools/nsc/doc/DocFactory.scala [hunk 1]
```diff
@@ -32,25 +32,23 @@ import DocParser.Parsed
...
-    override protected def computeInternalPhases() {
-      phasesSet += syntaxAnalyzer
-      phasesSet += analyzer.namerFactory
-      phasesSet += analyzer.packageObjects
-      phasesSet += analyzer.typerFactory
-      phasesSet += superAccessors
-      phasesSet += pickler
-      phasesSet += refChecks
+  def mkCompiler = 
+    new Global(settings, reporter) with interactive.RangePositions {
```
Comment (hubertp) [inline src/compiler/scala/tools/nsc/doc/DocFactory.scala:36]: the title of the commit is misleading. For me it suggests that we should have new interactive.Global(....)..... or am I missing something?

**Comment by vigdorchik** [comment]
@VladUreche right, I mean forInteractive implies forScaladoc.

**Comment by vigdorchik** [comment]
@hubertp Would "Use presentation compiler when scaladocing.in the IDE" be clearer?

**Comment by hubertp** [comment]
ah, now I get it. You guys probably talked about it so I am missing the context.
Can I know if there is any particular reason why you force scaladoc mode for _every_ usage of interactive.Global? In practice every IDE (not only Eclipse Scala IDE) extends interactive.Global and customizes it by overriding members so it is not obvious for me why we change the behaviour globally.

**Comment by hubertp** [comment]
Personally, I was just interested why you need to have it. There are things like Repl in interactive package which probably don't really want to typecheck docs (not that anyone would write it but still). I'm just not a fun of setting some values implicitly and globally.

Finally we have onlyPresentation def in nsc.Global which basically combined forScaladoc and forInteractive so I am being picky here to make sure that what you change here is not a mistake.

**Comment by hubertp** [comment]
Thanks, that looks better, I think.

**Comment by VladUreche** [comment]
LGTM for the latest commit.

**Pull request closed without merging**